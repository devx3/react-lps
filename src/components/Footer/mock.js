export const mock = {
    html: 'Thanks little bastard. <a href="http://localhost">DevX3</a>.',
};
