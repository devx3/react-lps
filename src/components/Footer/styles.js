import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
    text-align: center;
    ${({ theme }) => css`
        border-top: 1px solid ${theme.colors.mediumGrey};
        a {
            text-decoration: none;
        }

        & ${TextComponent} p {
            font-size: ${theme.fonts.sizes.xsmall};
        }

        & ${SectionContainer} {
            padding-top: 0;
            padding-bottom: 0;
        }
    `};
`;
