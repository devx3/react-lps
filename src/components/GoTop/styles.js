import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        background: ${theme.colors.secondaryColor};
        color: ${theme.colors.primaryColor};
        display: flex;
        align-items: center;
        justify-content: center;
        position: fixed;
        width: 4rem;
        height: 4rem;
        bottom: 2rem;
        right: 2rem;
        z-index: 6;
        cursor: pointer;
    `}
`;
