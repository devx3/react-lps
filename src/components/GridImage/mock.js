export const mock = {
    title: 'Hello GridImage',
    description: `Veniam velit magna nulla ex veniam sunt est.
    Consectetur ea aliquip do in. Qui amet non mollit elit proident
    aute adipisicing labore cillum. Non labore commodo minim labore
    ad sint eiusmod mollit anim cillum culpa officia cupidatat
    Lorem. Duis excepteur non ut consectetur adipisicing veniam sint.
    Aliquip ex excepteur non deserunt Lorem. Eu excepteur irure
    labore aliquip non aute sint sunt elit laboris occaecat ad veniam.`,
    grid: [
        {
            altText: 'Image 1',
            srcImage: 'https://source.unsplash.com/random/800x800?r=1',
        },
        {
            altText: 'Image 2',
            srcImage: 'https://source.unsplash.com/random/800x800?r=2',
        },
        {
            altText: 'Image 3',
            srcImage: 'https://source.unsplash.com/random/800x800?r=3',
        },
        {
            altText: 'Image 4',
            srcImage: 'https://source.unsplash.com/random/800x800?r=4',
        },
        {
            altText: 'Image 5',
            srcImage: 'https://source.unsplash.com/random/800x800?r=5',
        },
        {
            altText: 'Image 6',
            srcImage: 'https://source.unsplash.com/random/800x800?r=6',
        },
    ],
};
