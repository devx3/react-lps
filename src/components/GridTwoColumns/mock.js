export const mock = {
    title: 'Hello GridTwoColumns',
    text: `Ipsum cillum qui duis duis dolor est. Ut eiusmod do enim labore.
    Dolore culpa exercitation dolore cupidatat exercitation elit est
    reprehenderit aute consectetur esse. Tempor aute qui minim ullamco
    dolore dolore qui. Ex labore reprehenderit duis enim incididunt irure.
    Aliqua laboris velit deserunt irure mollit excepteur ea.
    Fugiat minim ex non id officia quis.`,
    darkMode: true,
    imgSrc: 'assets/images/javascript.svg',
};
