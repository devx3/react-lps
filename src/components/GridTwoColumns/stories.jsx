import { GridTwoColumns } from '.';
import { mock } from './mock';

export default {
    title: 'GridTwoColumns',
    component: GridTwoColumns,
    args: mock,
    argTypes: {
        children: { type: 'string' },
    },
};

export const GridDark = (args) => {
    return (
        <div>
            <GridTwoColumns {...args} />
        </div>
    );
};

export const Grid = (args) => {
    return (
        <div>
            <GridTwoColumns {...args} darkMode={false} />
        </div>
    );
};
