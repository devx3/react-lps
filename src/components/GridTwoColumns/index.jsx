import * as Styled from './styles.js';
import P from 'prop-types';
import { SectionBackground } from '../SectionBackground';
import { Heading } from '../Heading';
import { TextComponent } from '../TextComponent';

const GridTwoColumns = ({
    title,
    text,
    imgSrc,
    darkMode = false,
    sectionId = '',
}) => {
    return (
        <SectionBackground background={darkMode} sectionId={sectionId}>
            <Styled.Container>
                <Styled.TextContainer>
                    <Heading uppercase darkMode={darkMode}>
                        {title}
                    </Heading>
                    <TextComponent>{text}</TextComponent>
                </Styled.TextContainer>
                <Styled.ImageContainer>
                    <Styled.Image src={imgSrc} alt={title} />
                </Styled.ImageContainer>
            </Styled.Container>
        </SectionBackground>
    );
};

GridTwoColumns.propTypes = {
    title: P.string.isRequired,
    text: P.string.isRequired,
    imgSrc: P.string.isRequired,
    darkMode: P.bool,
    sectionId: P.string,
};

export default GridTwoColumns;
