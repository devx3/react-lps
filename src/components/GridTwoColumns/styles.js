import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacings.large};

        @media ${theme.media.ltMedia} {
            grid-template-columns: 1fr;
        }
    `}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css`
        @media ${theme.media.ltMedia} {
            margin-bottom: ${theme.spacings.large};
        }

        ${Heading} {
            margin-bottom: ${theme.spacings.large};
        }
    `}
`;

export const ImageContainer = styled.div`
    ${() => css``}
`;

export const Image = styled.img`
    ${() => css``}
`;
