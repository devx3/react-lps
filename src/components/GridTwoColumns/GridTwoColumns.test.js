import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import GridTwoColumns from '.';
import { mock } from './mock';

describe('<GridTwoColumns />', () => {
    it('Should render two columns grid', () => {
        const { container } = renderTheme(<GridTwoColumns {...mock} />);
        expect(container).toMatchSnapshot();
    });
    it('Should render two columns grid darkMode', () => {
        const { container } = renderTheme(
            <GridTwoColumns {...mock} darkMode={undefined} />,
        );
        expect(container).toMatchSnapshot();
    });
});
