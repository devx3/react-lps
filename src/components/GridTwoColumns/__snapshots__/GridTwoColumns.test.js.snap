// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> Should render two columns grid 1`] = `
.c5 {
  color: #FFFFFF;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c3 .c4 {
  margin-bottom: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem !important;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c6 {
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
    id="id-8664-94456859749"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Hello GridTwoColumns
          </h1>
          <div
            class="c6"
          >
            Ipsum cillum qui duis duis dolor est. Ut eiusmod do enim labore.
    Dolore culpa exercitation dolore cupidatat exercitation elit est
    reprehenderit aute consectetur esse. Tempor aute qui minim ullamco
    dolore dolore qui. Ex labore reprehenderit duis enim incididunt irure.
    Aliqua laboris velit deserunt irure mollit excepteur ea.
    Fugiat minim ex non id officia quis.
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Hello GridTwoColumns"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridTwoColumns /> Should render two columns grid darkMode 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c3 .c4 {
  margin-bottom: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem !important;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c6 {
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
    id="id-6490-447046366592"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Hello GridTwoColumns
          </h1>
          <div
            class="c6"
          >
            Ipsum cillum qui duis duis dolor est. Ut eiusmod do enim labore.
    Dolore culpa exercitation dolore cupidatat exercitation elit est
    reprehenderit aute consectetur esse. Tempor aute qui minim ullamco
    dolore dolore qui. Ex labore reprehenderit duis enim incididunt irure.
    Aliqua laboris velit deserunt irure mollit excepteur ea.
    Fugiat minim ex non id officia quis.
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Hello GridTwoColumns"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
